core/src/iotcl_telemetry.c

```diff
@@ -140,7 +144,17 @@ static int iotcl_telemetry_set_functions_common(
 
         // truncate the duplicate here so we can nest with the object name
         object_name[dot_index] = '\0';
-        *parent_object = cJSON_AddObjectToObject(message->current_data_set, object_name);
+
+        cJSON *parent_obj_ptr = cJSON_GetObjectItem(message->current_data_set, object_name);
+        if (parent_obj_ptr) {
+            if (!cJSON_IsObject(parent_obj_ptr)) {
+                IOTCL_ERROR(IOTCL_ERR_BAD_VALUE, "%s: Error: \"%s\" must be an object type and not a value!", function_name, object_name);
```

Comment:
Should this say something like '%s is an unknown parent object'? Maybe it's misleading otherwise

Reply:
We DO find the parent, so the parent element exists, but it is not an object. For example someone says

telemetry_set_number("coordinate", 2)
but then

telemetry_set_number("coordinate.x", 3)
We already found the intended parent coordinate, but it's a number and not an object, so we can't add "x" to it.
It would be a rare case, but I am testing for it in the unit test. I couldn't think of a better errror message though, but it's not that the parent is missing. If it's missing, then no problem - we add a new one in the else line below.